.github/workflows/rust.yml

```diff
@@ -153,3 +153,25 @@ jobs:
           name: fedimint
           authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
       - run: nix build
+  check-unused-dependencies:
+    runs-on: buildjet-8vcpu-ubuntu-2004
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-check-unused-dependencies-${{ hashFiles('**/Cargo.toml') }}
```

Comment:
What does all this stuff do? Seems like we should just keep using `Swatinem/rust-cache@v1` like we do in all the other jobs.

Reply:
Is it because this needs nightly?

Reply:
No it's not because of nightly its just how they used [cache](https://github.com/actions/cache) and I thought it was reasonable but unsure whether I should keep it:
```
Inputs

    path - A list of files, directories, and wildcard patterns to cache and restore. See [@actions/glob](https://github.com/actions/toolkit/tree/main/packages/glob) for supported patterns.
    key - An explicit key for restoring and saving the cache
    restore-keys - An ordered list of keys to use for restoring stale cache if no cache hit occurred for key. Note cache-hit returns false in this case.
```
I suppose the things in `path` are all used/touched by `udeps` and that's why they're stated there.
Then it creates the key by hashing the Cargo.toml which makes sense, but idk what we'd use that key for later

I'll change it to `Swatinem/rust-cache@v1`

Reply:
*hashing all `Cargo.tomls`